### ISSUE TITLE
Intercepting requests does not intercept automatic authentication requests

### ISSUE THREAD

**Issue by aine-rb** [zaproxy/zaproxy#2554]
### What steps will reproduce the problem?
1. Choose a test site. For this example, I'll use WebGoat
2. Create a context with an automatic Authentication method, for example Form-based Authentication
3. Configure the Authentication method
5. Use the User Forced Mode to send authentication request when needed
6. Click on the "Set break on all requests and responses" button

### What is the expected output? What do you see instead?
When authentication is needed, the authentication request is sent (it's shown in the history tab). However, the ZAP request interceptor can't intercept this request.
When authenticating manually (directly through the web browser), the authentication request is well intercepted by the ZAP interceptor.
Maybe it's a wanted/normal/conception-resulting behavior. However, I think it can be useful to be able to break every single request coming from or passing through ZAP.

### What version of the product are you using?
2.4.3

**Comment by psiinon** [comment]
It doesnt surprise me that this happens, but I also agree that it would be really useful to be able to intercept and change such calls :)

**Comment by thc202** [comment]
Right, that's the expected behaviour for all automatic/programmatic requests (authentication, spider, active scanner...).

Should we allow to change just (automatic) authentication requests or all types?

**Comment by aine-rb** [comment]
I think it's important to do this for authentication (the interceptor can be useful for debugging authentication scripts and understanding what is truly sent to the server).
However, I guess it would be a problem to allow this behaviour for automatic scans like the Spider one. These scans send numerous requests at a time. Intercepting these requests would prevent the user to use the ZAP interceptor efficiently for anything else in the meantime, in addition to decrease drastically the time performance of the current scans.

**Comment by thc202** [comment]
I was thinking in allowing to choose which types are intercepted, so, as debugging/helping functionality those (scanner) types would be disabled most of the time.
(Note that the same problem applies to authentication requests, it would "hang" the scanners while intercepting the (re)authentication requests.)

**Comment by aine-rb** [comment]
Hum i think your idea is the best :)
